pandas/tests/indexing/test_loc.py

```diff
@@ -630,3 +630,12 @@ def test_loc_empty_list_indexer_is_ok(self):
         tm.assert_frame_equal(df.loc[[]], df.iloc[:0, :],
                               check_index_type=True,
                               check_column_type=True)
+
+    def test_loc_identity_slice_returns_new_object(self):
+        # GH13873
+        df = DataFrame({'a': [1, 2, 3]})
+        result = df.loc[:]
```

Comment:
also can you add a couple of tests (can be in the same function is ok) that assert that things are NOT a copy (
df.iloc[:, 0] for example (IOW want to test the counter-examples)